scripts/package_windows_conpty.py

```
@@ -0,0 +1,242 @@
+        element = metadata.find(f"n:{field}", namespace)
+        actual = element.text.strip() if element is not None and element.text else None
+        if actual != expected_value:
+            raise ValueError(
+                f"package NuSpec {field} mismatch: expected {expected_value!r}, got {actual!r}"
+            )
+
+
+def acquire_package(package: dict[str, Any], package_path: Path) -> None:
+    package_path.parent.mkdir(parents=True, exist_ok=True)
+    if not package_path.exists():
+        with urllib.request.urlopen(package["url"]) as response, package_path.open("wb") as output:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)